### PR TITLE
 Fix storageEndpoint suffix for ADS policy

### PR DESCRIPTION
**Pull request by ayeletshpigelman**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Sql/Sql/AdvancedThreatProtection/Cmdlet/DisableAzureSqlManagedInstanceAdvancedDataSecurity.cs [hunk 1]
```diff
@@ -31,7 +32,7 @@ public class DisableAzureSqlManagedInstanceAdvancedDataSecurity : SqlManagedInst
         protected override ManagedInstanceAdvancedDataSecurityPolicyModel PersistChanges(ManagedInstanceAdvancedDataSecurityPolicyModel model)
         {
             model.IsEnabled = false;
-            ModelAdapter.SetManagedInstanceAdvancedDataSecurity(model);
+            ModelAdapter.SetManagedInstanceAdvancedDataSecurity(model, DefaultContext.Environment.GetEndpoint(AzureEnvironment.Endpoint.StorageEndpointSuffix));
```
Comment (markcowl) [inline src/Sql/Sql/AdvancedThreatProtection/Cmdlet/DisableAzureSqlManagedInstanceAdvancedDataSecurity.cs]: Do we have test coverage for this?  Also, please update the changelog in src\Sql\Sql  for this

**Comment by markcowl** [comment]
@yaakoviyun @jaredmoo CFan one of you take a look?

File: src/Sql/Sql/AdvancedThreatProtection/Cmdlet/EnableAzureSqlManagedInstanceAdvancedDataSecurity.cs [hunk 2]
```diff
@@ -59,7 +60,7 @@ protected override ManagedInstanceAdvancedDataSecurityPolicyModel PersistChanges
 
             if (DoNotConfigureVulnerabilityAssessment)
             {
-                ModelAdapter.SetManagedInstanceAdvancedDataSecurity(model);
+                ModelAdapter.SetManagedInstanceAdvancedDataSecurity(model, DefaultContext.Environment.GetEndpoint(AzureEnvironment.Endpoint.StorageEndpointSuffix));
```
Comment (tal12586) [inline src/Sql/Sql/AdvancedThreatProtection/Cmdlet/EnableAzureSqlManagedInstanceAdvancedDataSecurity.cs]: Why not add `DefaultContext.Environment.GetEndpoint(AzureEnvironment.Endpoint.StorageEndpointSuffix)`  to SqlAdvancedDataSecurityAdapter.cs and then you can remove the extra parameter from `SetManagedInstanceAdvancedDataSecurity`.

Like this:
`SqlThreatDetectionAdapter.SetManagedInstanceThreatDetectionPolicy(threatDetectionPolicy, DefaultContext.Environment.GetEndpoint(AzureEnvironment.Endpoint.StorageEndpointSuffix));`

Then you just have to change one file :)

Comment (talhers) [reply src/Sql/Sql/AdvancedThreatProtection/Cmdlet/EnableAzureSqlManagedInstanceAdvancedDataSecurity.cs]: This comment is by me, I accidentally used my other account :)

File: src/Sql/Sql/AdvancedThreatProtection/Cmdlet/EnableAzureSqlServerAdvancedDataSecurity.cs [hunk 3]
```diff
@@ -77,7 +78,7 @@ protected override ServerAdvancedDataSecurityPolicyModel PersistChanges(ServerAd
                 }
                 else
                 {
-                    ModelAdapter.SetServerAdvancedDataSecurity(model);
+                    ModelAdapter.SetServerAdvancedDataSecurity(model, DefaultContext.Environment.GetEndpoint(AzureEnvironment.Endpoint.StorageEndpointSuffix));
```
Comment (talhers) [inline src/Sql/Sql/AdvancedThreatProtection/Cmdlet/EnableAzureSqlServerAdvancedDataSecurity.cs]: Why not add `DefaultContext.Environment.GetEndpoint(AzureEnvironment.Endpoint.StorageEndpointSuffix)` to SqlAdvancedDataSecurityAdapter.cs and then you can remove the extra parameter from `SetManagedInstanceAdvancedDataSecurity`.

Like this:
`SqlThreatDetectionAdapter.SetManagedInstanceThreatDetectionPolicy(threatDetectionPolicy, DefaultContext.Environment.GetEndpoint(AzureEnvironment.Endpoint.StorageEndpointSuffix));`

Then you just have to change one file

**Comment by ayeletshpigelman** [comment]
this fix is included in another pull request: https://github.com/Azure/azure-powershell/pull/9371/files

**Pull request closed without merging**